timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -427,16 +417,26 @@ void __global__ k_nonbonded_unified(
                 coords,
                 params,
                 shared_box,
+                block_energy_buffer,
                 beta,
                 cutoff,
                 row_idxs,
                 ixn_tiles,
                 ixn_atoms,
                 du_dx,
-                du_dp,
-                u_buffer,
-                u_overflow_count);
+                du_dp);
         };
         tile_idx += stride;
     }
+    if (COMPUTE_U) {
+        // Sync to ensure the shared buffers are populated
+        __syncthreads();
+        if (threadIdx.x == 0) {
+            __int128 accumulated_energy = 0;
+            for (int i = 0; i < THREADS; i++) {
```

Comment:
We should definitely parallelize this as well using the same strategy.

Reply:
Refactored to use the same logic here: https://github.com/proteneer/timemachine/pull/1090/commits/f335db89264349c37d13f05af03223c5b033bef4